packages/babel-preset-udemy-website/raw.js

```diff
@@ -109,6 +109,16 @@ module.exports = {
         // ES2015 generators and ES2017 async functions (transform)
         ['transform-regenerator', { generators: true, async: true, asyncGenerators: false }],
 
+        // babel-plugin-react-css-modules
+        ['react-css-modules', {
+            filetypes: {
+                '.less': {
+                    syntax: 'postcss-less',
+                },
+            },
+            generateScopedName: '[name]--[local]--[hash:base64:5]',
```

Comment:
Does this need to be same pattern with the webpack olugins configuration? If so can we add a comment about that fact on both places?

Reply:
Oh perhaps I am confused and that plugin is no more?

Reply:
Btw how is this able to tell .global.less should not have isolated css rules, vs .less should?

Reply:
> Does this need to be same pattern with the webpack olugins configuration? If so can we add a comment about that fact on both places?

Thanks for helping me understand that! I didn't know we actually had that configured. I was just matching the previous behavior. I'll put a comment on both sides.

Reply:
> Btw how is this able to tell .global.less should not have isolated css rules, vs .less should?

I only understand this 80%. This is only for React components. It's not used for *.global.less files that get compiled into standalone .css files.

CC @kevinzang

Reply:
@jjinux @cansin My understanding is babel-plugin-react-css-modules currently compiles all .less files, including .global.less files, but this is irrelevant in terms of ".global.less should not have isolated css rules, vs .less should" because the plugin is not generating any CSS- that's still being done by the [less loaders](https://github.com/udemy/website-django/blob/master/static/webpack/helpers/less-loaders.js). The plugin's only job is to transform `styleName` to `className` in JSX files.

As an example, if we rename merchandising-course-card.less to merchandising-course-card.global.less, the plugin still works, e.g. it'll generate things like `class="merchandising-course-card-global--card--4pWMi"`, but the visuals are broken because the less loaders still generate global CSS, e.g. `.card` instead of `.merchandising-course-card-global--card--4pWMi`.

Although the current configuration does not break anything, I expect the plugin to be faster if we add the `"exclude": "\\.global\\.less$"` option, since it'll compile fewer .less files. In practice I don't think we ever pair `styleName` with selectors in .global.less files, but I have not tested the site thoroughly with this change.

Reply:
@kevinzang, I'm surprised it touches any .less file that isn't imported in JavaScript. Are you sure?

Reply:
@jjinux many base-components import .global.less files, for example https://github.com/udemy/website-django/blob/master/static/src/udemy/js/base-components/button.react-component.js#L6.

Reply:
Ah thanks kevin. What you’re saying makes sense. I couldn’t have figured it out 😊

Reply:
And +1 for excluding .global.less files.

Reply:
Ok.